src/main/java/org/apache/zeppelin/interpreter/SnappyDataZeppelinInterpreter.java

```diff
@@ -725,93 +733,119 @@ public InterpreterResult interpret(String[] lines, InterpreterContext context) {
     }
   }
 
+  private void addDeployedJars() {
+    if (snappyGlobalLoader != null) {
+      URL[] allUrls = snappyGlobalLoader.getURLs();
+      if (addedURLS == null || (addedURLS.length != allUrls.length)) {
+        List<URL> added = new ArrayList<>();
+        for (URL url : allUrls) {
+          try {
+            sc.addJar(url.getFile());
+            added.add(url);
+          } catch (Throwable t) {
+            // ignore may be previously added
+          }
+        }
+        addedURLS = added.toArray();
```

Comment:
What's the variable 'addedURLS' used for?

Reply:
Added urls is kind of a cache to see whether we have already added all the jars or not. There can be a case where a deployment has happened after the interpreter has come up in which case it needs to add the jars again. Otherwise it is not required.